addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -521,6 +522,84 @@ public void loadRssiValues() throws IOException {
         }
     }
 
+    @Override
+    public void setInstallMode(boolean enable, int seconds) throws IOException {
+        HmDevice gwExtrasHm = devices.get(HmDevice.ADDRESS_GATEWAY_EXTRAS);
+        
+        if (gwExtrasHm != null) {
+            // since the homematic virtual device exist: try setting install mode via its dataPoints
+            HmDatapoint installModeDataPoint = null;
+            HmDatapoint installModeDurationDataPoint = null;
+
+            // collect virtual datapoints to be accessed
+            for (HmChannel hmChannel : gwExtrasHm.getChannels()) {
+                HmDatapointInfo installModeDurationDataPointInfo = new HmDatapointInfo(HmParamsetType.VALUES, hmChannel,
+                        HomematicConstants.VIRTUAL_DATAPOINT_NAME_INSTALL_MODE_DURATION);
+                if (enable && hmChannel.hasDatapoint(installModeDurationDataPointInfo)) {
```

Comment:
You don't need the hmChannel.hasDatapoint, because hmChannel.getDatapoint returns null if the datapoint does not exist.

Reply:
Correct, since some lines later `if (installModeDurationDataPoint != null)` is checked anyways, `hasDatapoint(..)` can be removed here